client/a8c-for-agencies/components/agency-site-tags/index.tsx

```diff
@@ -1,64 +1,26 @@
-import { Button, Card } from '@automattic/components';
-import { useTranslate } from 'i18n-calypso';
-import { KeyboardEvent, useState } from 'react';
-import AgencySiteTag from 'calypso/a8c-for-agencies/components/agency-site-tag';
-import FormTextInput from 'calypso/components/forms/form-text-input';
-import './style.scss';
+import { FormTokenField } from '@wordpress/components';
+import { translate } from 'i18n-calypso';
+import { useState } from 'react';
 
 interface Props {
 	tags: string[];
 	isLoading: boolean;
-	onAddTags: ( newTags: string[] ) => void;
-	onRemoveTag: ( removeTag: string ) => void;
+	onSetTags: ( newTags: string[] ) => void;
 }
 
-export default function AgencySiteTags( { tags, isLoading, onAddTags, onRemoveTag }: Props ) {
-	const translate = useTranslate();
-	const [ tagsInput, setTagsInput ] = useState( '' );
+export default function AgencySiteTags( { tags: initialTags, onSetTags }: Props ) {
```

Comment:
If changes to the `tags` prop do not go through then the prop should be called initialTags to begin with.
Ideally, whatever changes we trigger here should go up the chain and down again resulting in the tags prop changing so we don't have to keep tags in local state as well.